lib/users_on_holiday.rb

```diff
@@ -5,17 +5,72 @@ class UsersOnHoliday
     def self.from(calendar_events)
       calendar_events
         .filter { |e| e.user_id.present? && e.username.present? }
-        .filter { |e| e.underway? }
+        .filter { |e| e.underway? || e.in_future? }
         .group_by(&:user_id)
-        .map { |_, events| events.sort_by(&:ends_at).last }
-        .to_h { |e| [
-          e.user_id,
-          {
-            username: e.username,
-            ends_at: e.ends_at
-          }
-        ]
+        .map { |_, events| current_holiday(events) }
+        .compact
+        .to_h
+    end
+
+    private
+
+    def self.current_holiday(user_events)
+      holiday_ends_at = holiday_ends_at(user_events)
```

Comment:
I usually tend to avoid naming variables with the same name as functions. How about this instead?

```suggestion
      ends_at = holiday_ends_at(user_events)
```